base/v0_2_exp/translate.go

```diff
@@ -100,6 +107,53 @@ func translateResource(from Resource, options base.TranslateOptions) (to types.R
 	to.Source = from.Source
 	to.Compression = from.Compression
 	tm.AddIdentity("source", "compression")
+
+	if from.Local != nil {
+		c := path.New("yaml", "local")
+
+		if options.FilesDir == "" {
+			r.AddOnError(c, ErrNoFilesDir)
+			return
+		}
+
+		// calculate file path within FilesDir and check for
+		// path traversal
+		filePath := filepath.Join(options.FilesDir, *from.Local)
+		absBasePath, err := filepath.Abs(options.FilesDir)
+		if err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+		absFilePath, err := filepath.Abs(filePath)
+		if err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+		if !strings.HasPrefix(absFilePath, absBasePath+string(filepath.Separator)) {
+			r.AddOnError(c, ErrFilesDirEscape)
+			return
+		}
+
+		infile, err := os.Open(filePath)
+		if err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+		defer infile.Close()
+		contents, err := ioutil.ReadAll(infile)
```

Comment:
Is there a particular reason to be opening the file manually and performing a `ReadAll` rather than a `ReadFile` call?

Reply:
Nope.  I haven't written Go in a while.  :slightly_smiling_face: